core/metrics-core-impl/src/main/java/org/hawkular/metrics/core/impl/CreateTenants.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright 2014-2015 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.hawkular.metrics.core.impl;
+
+import java.util.concurrent.CountDownLatch;
+
+import org.hawkular.metrics.tasks.api.Task2;
+import org.hawkular.metrics.tasks.api.Trigger;
+import org.joda.time.DateTime;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.datastax.driver.core.ResultSet;
+
+import rx.Observable;
+import rx.functions.Action1;
+
+/**
+ * @author jsanda
+ */
+public class CreateTenants implements Action1<Task2> {
+
+    private static final Logger logger = LoggerFactory.getLogger(CreateTenants.class);
+
+    public static final String TASK_NAME = "create-tenants";
+
+    private TenantsService tenantsService;
+
+    private DataAccess dataAccess;
+
+    public CreateTenants(TenantsService tenantsService, DataAccess dataAccess) {
+        this.tenantsService = tenantsService;
+        this.dataAccess = dataAccess;
+    }
+
+    @Override
+    public void call(Task2 task) {
+        long bucket = getBucket(task.getTrigger());
+        CountDownLatch latch = new CountDownLatch(1);
```

Comment:
What's the reason for blocking the task executor thread until an error occurs or tenant bucket is deleted?